R/template.R

```diff
@@ -77,6 +73,244 @@ metric_summarizer <- function(metric_nm,
                               ...,
                               metric_fn_options = list()) {
+  signal_soft_deprecated(
+    paste(
+      "`metric_summarizer()` has been soft-deprecated as of version 1.2.0.",
+      "Please see `?metric_summarizer()` for further instructions."
+      )
+  )
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+  case_weights <- enquo(case_weights)
+
+  validate_not_missing(truth, "truth")
+  validate_not_missing(estimate, "estimate")
+
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  finalize_estimator_expr <- rlang::expr(
+    finalize_estimator(!! truth, estimator, metric_nm)
+  )
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = metric_nm,
+    .estimator = eval_tidy(finalize_estimator_expr),
+    .estimate = metric_fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      !!! spliceable_estimator(estimator),
+      na_rm = na_rm,
+      !!! spliceable_event_level(event_level),
+      !!! spliceable_case_weights(case_weights),
+      !!! metric_fn_options
+    )
+  )
+
+  dplyr::as_tibble(metric_tbl)
+}
+
+#' Developer function for summarizing new numeric metrics
+#'
+#' `metric_summarizer_numeric()` is useful alongside [metric_vec_template()] for
+#' implementing new custom metrics. `metric_summarizer_numeric()` calls the
+#' metric function inside `dplyr::summarise()`. `metric_vec_template()` is a
+#' generalized function that calls the core implementation of a metric function,
+#' and includes a number of checks on the types, lengths, and argument inputs.
+#' See [Custom performance
+#' metrics](https://www.tidymodels.org/learn/develop/metrics/) for more
+#' information.
+#'
+#' @details
+#'
+#' `metric_summarizer_numeric()` is generally called from the data frame version
+#' of your metric function. It knows how to call your metric over grouped data
+#' frames and returns a `tibble` consistent with other metrics.
+#'
+#'
+#' @inheritParams metric_summarizer
+#' @param metric_nm A single character representing the name of the metric to
+#' use in the `tibble` output.
+#'
+#' @param estimate Generally, the unquoted column name corresponding to
+#' the `estimate` column.
+#'
+#' @seealso [metric_vec_template()] [finalize_estimator()] [dots_to_estimate()]
+#'   [metric_summarizer_class()] [metric_summarizer_prob()]
+#'
+#' @export
+metric_summarizer_numeric <- function(metric_nm,
+                                      metric_fn,
+                                      data,
+                                      truth,
+                                      estimate,
+                                      na_rm = TRUE,
+                                      case_weights = NULL,
+                                      ...,
+                                      metric_fn_options = list()) {
+  truth <- enquo(truth)
```

Comment:
Time has shown that we _aren't_ going to use the `...` here, so I think we should force all optional arguments to be named arguments by moving the `...` up.

This is also a nice time to use `rlang::check_dots_empty()`

```suggestion
                                      ...,
                                      na_rm = TRUE,
                                      case_weights = NULL,
                                      metric_fn_options = list()) {
  rlang::check_dots_empty()
  
  truth <- enquo(truth)
```

When you work on the documentation for these functions again, you can use `@inheritParams rlang::args_dots_empty` and that will give you consistent documentation for empty `...`

Reply:
Done in https://github.com/tidymodels/yardstick/pull/322/commits/fb0e2823c20e1a30a18f26ec6bd93291c5efdc59